tests/v2/test_waku_store.nim

```diff
@@ -377,3 +377,84 @@ procSuite "Waku Store":
       decodedEmptyRes.isErr == false
       decodedEmptyRes.value == emptyRes
     
+  asyncTest "temporal history queries":
+    let
+      key = PrivateKey.random(ECDSA, rng[]).get()
+      peer = PeerInfo.init(key)
+    var
+      msgList = @[WakuMessage(payload: @[byte 0], contentTopic: ContentTopic(2)),
+        WakuMessage(payload: @[byte 1],contentTopic: ContentTopic(1), timestamp: float(1)),
+        WakuMessage(payload: @[byte 2],contentTopic: ContentTopic(2), timestamp: float(2)),
+        WakuMessage(payload: @[byte 3],contentTopic: ContentTopic(1), timestamp: float(3)),
+        WakuMessage(payload: @[byte 4],contentTopic: ContentTopic(2), timestamp: float(4)),
+        WakuMessage(payload: @[byte 5],contentTopic: ContentTopic(1), timestamp: float(5)),
+        WakuMessage(payload: @[byte 6],contentTopic: ContentTopic(2), timestamp: float(6)),
+        WakuMessage(payload: @[byte 7],contentTopic: ContentTopic(1), timestamp: float(7)),
+        WakuMessage(payload: @[byte 8],contentTopic: ContentTopic(2), timestamp: float(8)),
+        WakuMessage(payload: @[byte 9],contentTopic: ContentTopic(1),timestamp: float(9))]
+            
+    var dialSwitch = newStandardSwitch()
+    discard await dialSwitch.start()
+
+    var listenSwitch = newStandardSwitch(some(key))
+    discard await listenSwitch.start()
+
+    let
+      proto = WakuStore.init(PeerManager.new(dialSwitch), crypto.newRng())
+      subscription = proto.subscription()
+    proto.setPeer(listenSwitch.peerInfo)
+
+    var subscriptions = newTable[string, MessageNotificationSubscription]()
+    subscriptions["test"] = subscription
+
+    listenSwitch.mount(proto)
+
+    for wakuMsg in msgList:
+      await subscriptions.notify("foo", wakuMsg)
+    
+    asyncTest "handle temporal history query with a valid time window":
+      var completionFut = newFuture[bool]()
+
+      proc handler(response: HistoryResponse) {.gcsafe, closure.} =
+        check:
+          response.messages.len() == 2
+          response.messages.anyIt(it.timestamp == float(3))
+          response.messages.anyIt(it.timestamp == float(5))
+        completionFut.complete(true)
+
+      let rpc = HistoryQuery(topics: @[ContentTopic(1)], startTime: float(2), endTime: float(5))
+      await proto.query(rpc, handler)
+
+      check:
+        (await completionFut.withTimeout(5.seconds)) == true
+
+    asyncTest "handle temporal history query with a zero-size time window":
+      var completionFut = newFuture[bool]()
+
+      proc handler(response: HistoryResponse) {.gcsafe, closure.} =
+        check:
+          response.messages.len() == 5
+        completionFut.complete(true)
+
+      let rpc = HistoryQuery(topics: @[ContentTopic(1)], startTime: float(2), endTime: float(2))
```

Comment:
I may be missing something, but shouldn't the special case here be to only return messages with timestamp = startTime = endTime = float(2)? It seems a bit strange to me to return all historical messages if the window size is 0.

Reply:
I considered a zero-size time window as an invalid query (i.e., no temporal filter should take place). Somehow it did not make sense that someone sends a request whose response is already empty! Indeed, this is equivalent to leaving the time window unspecified wherein the window size is also zero and means that no time filter is requested.
However, in this case, I am assuming that the requester checks its window size prior to query which might not be true.
Conclusion: I am going to change this logic, and adopt your suggested approach, it would be safer. Thanks!